### ISSUE TITLE
peakbolt - USDC blocklist will prevent full liquidation of JUSD debt

### ISSUE THREAD

**Issue by sherlock-admin** [sherlock-audit/2023-04-jojo-judging#301]
peakbolt

medium

# USDC blocklist will prevent full liquidation of JUSD debt

## Summary

A user could use a USDC blocked address to borrow JUSD and prevent full liquidation of the JUSD debt. 

## Vulnerability Detail

During a liquidation, `FlashLoanLiquidate.JOJOFlashLoan()` will transfer `liquidateData.liquidatedRemainUSDC` to the liquidated address. This is to return the excess collateral in USDC to the liquidated user. 

The issue is that this transfer could fail if the liquidated address is blocked by USDC. That will cause the entire liquidation to fail.

USDC has a contract level admin controlled address blocklist, which will block transfer to/from these blocked addresses.

### Proof of Concept
In `JUSDV1`, add the following code to `"@JOJO/contracts/testSupport/TestERC20.sol"` to simulate a transfer block on alice.

```Solidity
    function transfer(address to, uint256 amount) public override returns (bool) {
        if(to == address(1)) return false; 
        return super.transfer(to, amount);
    }
```

Then run `testLiquidateAll()` in `JUSDBankOperatorLiquidate.t.sol`. The liquidation will revert on transfer of `liquidatedRemainUSDC` to alice (liquidated).

## Impact
As the protocol is not able to fully liquidate such USDC blocked addresses, it will incur bad debts and suffer losses.

## Code Snippet

https://github.com/sherlock-audit/2023-04-jojo/blob/main/JUSDV1/src/Impl/flashloanImpl/FlashLoanLiquidate.sol#L86

## Tool used
Manual review

## Recommendation
Consider a pull withdrawal pattern instead of push.

Duplicate of #206
